diff --git a/node_modules/react-native-screens/ios/RNSScreen.mm b/node_modules/react-native-screens/ios/RNSScreen.mm
index 65c18f1..d6abdd9 100644
--- a/node_modules/react-native-screens/ios/RNSScreen.mm
+++ b/node_modules/react-native-screens/ios/RNSScreen.mm
@@ -928,7 +928,11 @@ - (BOOL)isTransparentModal
 - (void)invalidate
 {
   _controller = nil;
-  [_sheetsScrollView removeObserver:self forKeyPath:@"bounds" context:nil];
+  @try {
+    [_sheetsScrollView removeObserver:self forKeyPath:@"bounds" context:nil];
+  } @catch (NSException *exception) {
+    RCTLogWarn(@"[RNScreens] Tried to remove non-registered observer in invalidate: %@", exception);
+  }
 }
 
 #if !TARGET_OS_TV && !TARGET_OS_VISION
